polire/interpolate/natural_neighbors/natural_neighbors.py

```diff
@@ -187,9 +183,7 @@ def _predict(self, X):
                 if len(new)<3:
                     ## We need atleast a traingle to interpolate
                     ## Three new voronoi vertices form a triangle
-                    print("Can't interpolate because of unbounded Voronoi Partition")
-                    print("Data is ", X[index])
-                    result[index] = None
+                    result[index] = np.nan
```

Comment:
can we return average or something? or usually, nan is returned in case of natural neighbor interpolation?

Reply:
I am not sure tbh. The gridded values I am not sure if we are actually putting the values in the right place back (Does reshape() do the reverse of ravel() )?Point wise interpolation is correct, since I conputed the values manually. 

Regarding average depends. Ill try doing that ( I think point_region) must be helpful

Reply:
> Does reshape() do the reverse of ravel().
Yes, it does.

> I think point_region must be helpful.
I am sorry, what do mean by point_region?

Reply:
`point_regions` from the first comment in this thread was what I was referring to. I was thinking about using it to take the average at the`nan` points. Actually, the interpolated grid is incorrect. We(I) usually fill in the values in a row wise manner. Check IDW’s `_predict_grid` method.
The implementation that you had given does it in a column wise manner. We need to add a `X1 = X1.T` and `X2 = X2.T`, (If you think about it, it’s really the transpose of the final grid. But then, that would most likely messing with the base class and hence I am suggesting the above option. Can you check once more?

Reply:
> The implementation that you had given does it in a column-wise manner.

I ran this in my interpreter. Isn't this row-wise?
```bash
>>> import numpy as np
>>> a = np.ones((3,3))
>>> a[:,0:1] = np.zeros((3,1))
>>> a
array([[ 0.,  1.,  1.],
       [ 0.,  1.,  1.],
       [ 0.,  1.,  1.]])
>>> a.ravel()
array([ 0.,  1.,  1.,  0.,  1.,  1.,  0.,  1.,  1.])
>>> 
```

Reply:
This isn't the issue. The issue is with Meshgrid. Check how X1 and X2 are returned.

Take this as an example:
```python
>>>  import numpy as np
>>> x1 = np.linspace(1,10,10)
>>> x2 = np.linspace(1,5,5)
>>> X1, X2 = np.meshgrid(x1, x2)
>>> print(np.array([X1.ravel(), X2.ravel()]).T[:10])
```
Output is:
`[[ 1.  1.]
 [ 2.  1.]
 [ 3.  1.]
 [ 4.  1.]
 [ 5.  1.]
 [ 6.  1.]
 [ 7.  1.]
 [ 8.  1.]
 [ 9.  1.]
 [10.  1.]]`

Reply:
Our predict function predicts on this, and when you reshape, it is done in a row wise manner, effectively swapping out for a column. Do correct me if I am wrong,

Reply:
Sorry, I am not able to get your problem. Can you please confirm it with a nested loop outside the `predict` method and comparing if the returned vectors (from predict and predict_grid) are close or not?